package.yaml

```diff
@@ -39,6 +39,7 @@ default-extensions:
   - NamedFieldPuns
```

Comment:
> - add hlint rule for `Fmt.indentF` that suggests not to use functions generating lines of spaces

Agreed, good idea!

> - more docs/meaningful names to new functions?

:+1: 

> - currently `indentF_` is made "from scratch" inspired by `Fmt.identF` source, and `blockListF'_` depends on `Fmt.blockListF'`, and simply deletes lines containing only spaces from it's result.  Maybe it's better to use one approach (which one)?

I prefer the approach you used with `blockListF_`. Run the underlying Fmt function, iterate through all the lines, and replace all-white-space lines with `""`.

In fact, this could even be abstracted into some function of type `Builder -> Builder`, that we then reuse in both `indentF_` and `blockListF'_`

Reply:
An even more robust approach could be to just remove all trailing whitespace from every line (using `Text.stripEnd`)

Reply:
Is it enough to have only `interpolateIndentF` and `interpolateBlockList` functions that are same time not producing trailing whitespaces nor trailing newlines?

Reply:
> Is it enough to have only interpolateIndentF and interpolateBlockListF functions that are same time not producing trailing whitespaces nor trailing newlines?

Yeah, I think so.

I think we're only using it in the context of interpolation anyway.

> I would also prefer interpolateBlockListF to take non-empty instead of lists, btw

How would you handle empty lists (e.g., printing `_fiAnchors` in `Buildable FileInfo`)?

Reply:
Something like
```
case nonEmpty _fiAnchors of
  Just ne -> ...
  Nothing -> "-"
```

Reply:
Okay makes sense - different places will want to handle empty lists differently.

----

Alternatively, our version of `blockListF` could keep accepting normal lists, but also take an additional `Builder` parameter  to display when the list is empty.

Reply:
Done. Choosed version with NonEmty, since there was nontrivial "null checks" before printing blockLists. Also changed reports a bit - you  can see changes in bats tests. + `xrefcheck -v` is more informative when there is no scannable files:
Was
```
=== Repository data ===
  []
```

Now
```
=== Repository data ===
  No scannable files found.
```

Reply:
Great work, well done! :+1: